### PR TITLE
Add tests for crossdock echo behaviors

### PR DESCRIPTION
**Pull request by abhinav**
Resolves https://github.com/yarpc/yarpc-go/issues/33

### CODE REVIEW

**Comment by abhinav** [comment]
@yarpc/golang

File: crossdock/client/echo/behavior.go [hunk 1]
```diff
@@ -82,7 +83,8 @@ func createRPC(s behavior.Sink, p behavior.Params) yarpc.RPC {
 	trans := p.Param(TransportParam)
 	switch trans {
 	case "http":
-		outbound = http.NewOutbound(fmt.Sprintf("http://%s:8081", server))
+		cl := &http.Client{Transport: new(http.Transport)}
```
Comment (prashantv) [inline crossdock/client/echo/behavior.go]: nit: put a comment about why you use a custom client + transport here

File: crossdock/client/echo/behavior_test.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	"testing"
+
+	"github.com/yarpc/yarpc-go/crossdock/client/behavior"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEchoSink(t *testing.T) {
+	tests := []struct {
+		server    string
```
Comment (prashantv) [inline crossdock/client/echo/behavior_test.go]: optionally:
```go
server, transport, encoding string
```
matches the initialization of the struct that way too

Comment (abhinav) [inline crossdock/client/echo/behavior_test.go]: I wanted to preserve horizontal whitespace in the f/entry alignment

File: crossdock/client/echo/echo_test.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+			if tt.register != nil {
+				tt.register(rpc)
+			}
+
+			entries := behavior.Run(func(s behavior.Sink) {
+				err := rpc.Start()
+				assert.NoError(t, err,
+					"failed to start RPC for %v, %v", trans.name, tt.encname)
+				if err == nil {
```
Comment (prashantv) [inline crossdock/client/echo/echo_test.go]: nit: you can do
```go
if assert.NoError(t, err, "...") {
  defer rpc.Stop()
  // rest of the code
}
```

File: crossdock/client/echo/echo_test.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+				require.NoError(t, err, "could not create TChannel")
+				return tch.NewInbound(ch, tch.ListenAddr(":8082"))
+			},
+		},
+	}
+
+	tests := []struct {
+		encname  string
+		register func(yarpc.RPC)
+		// registerInvalid func(yarpc.RPC)
```
Comment (prashantv) [inline crossdock/client/echo/echo_test.go]: stale or expected to be used later?

**Comment by prashantv** [comment]
lgtm

File: crossdock/client/echo/behavior.go (hunk 1 above)
Comment (abhinav) [inline crossdock/client/echo/behavior.go]: :+1:

File: crossdock/client/echo/echo_test.go (hunk 3 above)
Comment (abhinav) [inline crossdock/client/echo/echo_test.go]: :+1:

File: crossdock/client/echo/echo_test.go (hunk 4 above)
Comment (abhinav) [inline crossdock/client/echo/echo_test.go]: Stale. Thanks.

**Comment by abhinav** [comment]
Updated

**Pull request merged**